plugins/org.jkiss.dbeaver.data.transfer/src/org/jkiss/dbeaver/tools/transfer/stream/exporter/DataExporterCSV.java

```diff
@@ -279,6 +285,10 @@ private void writeCellValue(String value, boolean quote)
         // check for needed quote
         final boolean hasQuotes = useQuotes && value.indexOf(quoteChar) != -1;
 
+        if (CommonUtils.isNotEmpty(lineFeedEscapeString)) {
```

Comment:
I actually have doubts about performance.
Now we will compile and match regex for every single cell. It means millions of regex compilations + matches for big files.
Although there may be zero matches.
It may slow down data export significantly.
Let's precompile pattern + verify that cell has at least one \n character.